src/MonoTorrent/MonoTorrent.Client/ClientEngine.cs

```diff
@@ -308,6 +308,7 @@ public async Task<bool> RemoveAsync (TorrentManager manager)
             manager.Engine = null;
             manager.DownloadLimiters.Remove (downloadLimiters);
             manager.UploadLimiters.Remove (uploadLimiters);
+            _ = manager.MaybeDeleteFastResumeAsync ();
```

Comment:
I had a task, which i forgot about, to address the issue of stale data.

I was wondering if this pair of methods would help:

```
// Iterate over all cache files. Delete any file which *does not* correspond to a torrent loaded in the engine.
ClientEngine.ClearStaleCacheFiles ();

// Deletes *all* cached fast resume data, dht nodes, etc etc etc
ClientEngine.ClearCacheFiles ();

// Deletes all fast resume data, or other torrent-specific data, for the supplied TorrentManager/MagnetLink/InfoHash
ClientEngine.ClearCacheFiles (InfoHash|TorrentManager|MagnetLink)
``` 

Usecases include:
```
async Task RemoveFromEngine (TorrentManager manager)
{
    // from the torrent from the engine
    await engine.RemoveAsync (manager);

    // Remove any cache files for torrents which are no longer in the engine
    await engine.ClearStaleCacheFiles ();

    // Or just remove the data for that torrent specifically:
    await engine.ClearCacheFiles (manager);
}
```

If something goes wrong at some point and you want to clear data before loading a torrent...
```
async Task AddToEngine (Torrent torrent)
{
    await engine.ClearCacheFiles (torrent.InfoHash);
    await engine.AddAsync (torrent);
}
``` 

Maybe there could be an enum too which lists the kinds of data which will be cleared, so you could selectively clear fast resume, or something else?